pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -115,3 +200,38 @@ func getInfraMachineFromProvider(platform configv1.PlatformType) (client.Object,
 		return nil, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
 	}
 }
+
+// shouldMirrorCAPIMachineToMAPIMachine takes a CAPI machine and determines if there should
+// be a MAPI mirror, it returns true only if:
+//
+// 1. The CAPI Machine is owned by a CAPI MachineSet,
+// 2. That CAPI MachineSet has a MAPI MachineSet Mirror.
+func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context.Context, machine *capiv1beta1.Machine, logger logr.Logger) (bool, error) {
+	// Check if owner refs point to CAPI MachineSet
+	for _, ref := range machine.ObjectMeta.OwnerReferences {
+		if ref.Kind != "MachineSet" || ref.APIVersion != "cluster.x-k8s.io/v1beta1" {
```

Comment:
Do we have constants we can use for this? We could use the group version stuff from `doc.go` in the CAPI api packages to start for API version

Reply:
I can't find any `doc.go` with the API version, other than maybe building it from: https://github.com/kubernetes-sigs/cluster-api/blob/main/api/v1beta1/groupversion_info.go#L30 ?

Maybe just a `const` at the top of the file?

Reply:
Ah sweet! got it